tests/clv/models/test_basic.py

```diff
@@ -1,80 +1,41 @@
 
 
 class CLVModelTest(CLVModel):
-    _model_name = "CLVModelTest"
+    _model_type = "CLVModelTest"
 
     def __init__(self):
         super().__init__()
-        with pm.Model(coords={"test_x_dim": [99]}) as self.model:
-            x = pm.HalfNormal("x", 100, dims=("test_x_dim",))
-            pm.Normal("y", x, observed=[1, 3, 3, 3, 5])
+
+    @property
+    def default_model_config(self):
+        pass
+
+    @property
+    def default_sampler_config(self):
+        pass
+
+
+@pytest.fixture(scope="module")
+def posterior():
+    # Create a random numpy array for posterior samples
+    posterior_samples = np.random.randn(
+        4, 100, 2
+    )  # shape convention: (chain, draw, *shape)
+
+    # Create a dictionary for posterior
+    posterior_dict = {"theta": posterior_samples}
+    return from_dict(posterior=posterior_dict)
 
 
 class TestCLVModel:
-    def test_fit(self):
-        model = CLVModelTest()
-
-        with pytest.raises(RuntimeError, match="The model hasn't been fit yet"):
-            model.fit_result
-
-        idata = model.fit(
-            tune=5,
-            chains=2,
-            draws=10,
-            compute_convergence_checks=False,
-        )
-        assert isinstance(idata, InferenceData)
-        assert len(idata.posterior.chain) == 2
-        assert len(idata.posterior.draw) == 10
-        assert model.fit_result is idata
-
-    def test_fit_MAP(self):
-        model = CLVModelTest()
-
-        idata = model.fit(fit_method="map")
-        assert isinstance(idata, InferenceData)
-        assert len(idata.posterior.chain) == 1
-        assert len(idata.posterior.draw) == 1
-        assert idata.posterior["x"].dims == ("chain", "draw", "test_x_dim")
-        assert model.fit_result is idata
-
-        # Check that summary only includes single value
-        summ = model.fit_summary()
-        assert isinstance(summ, Series)
-        assert summ.name == "value"
-
-    @patch("arviz.summary", return_value="fake_summary")
-    def test_fit_summary(self, dummy_summary):
-        model = CLVModelTest()
-        model._fit_result = "fake_fit"
-        fake_mcmc_fit = from_dict({"x": [[0, 0]]})
-        model._fit_result = fake_mcmc_fit
-        assert model.fit_summary(opt_kwarg="opt_kwarg") == "fake_summary"
-        dummy_summary.assert_called_once_with(fake_mcmc_fit, opt_kwarg="opt_kwarg")
-
-    def test_repr(self):
-        model = CLVModelTest()
-        assert (
-            model.__repr__() == "CLVModelTest\nx ~ HalfNormal(0, 100)\ny ~ Normal(x, 1)"
-        )
-
-    def test_fit_override(self):
-        with warnings.catch_warnings():
-            warnings.simplefilter("error")
-
-            model = CLVModelTest()
-            model.fit_result = 1
-            with pytest.warns(UserWarning, match="Overriding pre-existing fit_result"):
-                model.fit_result = 2
+
+    # The rest of the tests need to be moved to subclasses, as basic.py is not a standalone class
```

Comment:
Why not create a test subclass that implements the same kind of dummy model as was before? Then the shared methods can be tested here just like it was done so far

Reply:
I'm not a fan of tests where everything is mocked, they don't bring much security and are just confusing.

Reply:
If the point is to test methods implement in the base class it seems reasonable to implement the simplest subclass of the baseclass to test them

Reply:
It's fine to change how tests work but that's probably better suitable in a standalone PR than in a large refactor like this